en/references/tested-judge/template-description/README.md

```diff
@@ -0,0 +1,392 @@
+    The namespace is '${namespace_html}'.
+</p>
+```
+
+Instance JavaScript:
+```html
+<p>
+    The prompt for <span style="font-weight: bold">javascript</span> is <code>&gt;</code>.
+    The namespace is 'submission'.
+</p>
+```
+
+## Functions
+As already indicated in the section [Constants](#constants) Mako also supports functions.
+
+We will again given an overview of all available functions that could be used in the templates with some examples.
+
+- **function_name**:
+  This function is used the generated the correct style conventions for the function names.
+  The input of this function is one string.
+  ::: tip Hint
+  When you pass the function name in the template, it's best to use the `snake_case` style convention.
+  This makes it possible to generate the correct function names foreach language.
+  :::
+- **type_name**:
+  This function is used to translate the TESTed datatypes to the programming language datatypes.
+  The input of this function is either a string, or a pair as type argument.  
+  The first argument of the pair must be a collection datatype string,
+  the second either one type argument, or a list of type arguments.
+- **natural_type_name**:
+  This function searches the natural name for a collectiondatatype and/or the string-datatype of TESTed for a given
+  programming language.
+  The input of this function is one string.
+- **statement and expression**:
+  This functions are not meant to use by the redactor of the descriptions.
+  But for the sake of completeness, they are listed anyway.
+  This functions are used to convert code snippets in the mini-programming language of TESTed
+  (see [Statements, expressions and return-raw](../dsl/#statements-expressions-and-return-raw)) to code snippets in the
+  programming languages.
+  - **statement** must be used for all input code snippets.
+  - **expression** must be used for all return values.
+  
+  Both functions expect a string as argument.
+
+### Example
+Now we look at a Markdown example description for Python and JavaScript.
+
+Template:
+```mako
+The function ${function_name("characterize_string")} takes one
+${type_name("text")} as argument and returns the ${natural_type_name("sequence")}
+of characters of type ${type_name(("sequence", "char"))}.
+```
+
+Instance JavaScript:
+```markdown
+The function `characterizeString` takes one
+`string` as argument and returns the array
+of characters of type `array<string>`.
+```
+
+Instance Python:
+```markdown
+The function `characterize_string` takes one
+`str` as argument and returns the list
+of characters of type `List[str]`.
+```
+
+## Code examples
+An important possibility for the template descriptions, was the simplicity to notate the code examples.
+This is the only functionality that slightly differs between the Markdown and HTML version.
+The biggest difference is the way how the code blocks are represented.
+The content of these code blocks are the same for both markup languages.
```

Comment:
Is this something that should go into a technical reference? It looks more like something for a guide or thesis to me.